src/output/escape.rs

```diff
@@ -1,16 +1,37 @@
 use super::file_name::QuoteStyle;
+use crate::fs::File;
 use ansiterm::{ANSIString, Style};
 
+#[derive(Eq, Copy, PartialEq, Clone)]
+pub enum Quotes {
+    Single,
+    Double,
+}
+
+pub fn check_quote(files: &Vec<File<'_>>) -> Quotes {
+    for file in files {
```

Comment:
nit: 

&Vec<<_>> creates a new object, where you can just use a slice instead. (also, iterator is more intuitive Rust)
```rust
pub fn check_quote(files: &[File<'_>]) -> Quotes {
    if files.iter().any(|file| file.name.contains('\'')) {
        Quotes::Double
    } else {
        Quotes::Single
    }
}
```

Reply:
vec here is a ref to the vec used in main so it kinda made sense for me